ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/GenerateMissingRecordFieldsFix.fs

```diff
@@ -121,3 +112,139 @@ type GenerateMissingRecordFieldsFix(recordExpr: IRecordExpr) =
+                    else
+                        let anchor: ITreeNode =
+                            let indexedBinding = indexedBindings[fieldIndex - 1]
+                            if generateSingleLine then
+                                indexedBinding
+                            else
+                                x.SkipComments indexedBinding
+
+                        let resultingNode =
+                            // Nodes after block comments are not automatically moved to the new line, fixing it
+                            if (not generateSingleLine) && anchor.GetTokenType() == FSharpTokenType.BLOCK_COMMENT then
+                                let newLineNode = NewLine(binding.GetLineEnding())
+                                let insertedNewLine = ModificationUtil.AddChildAfter(anchor, newLineNode)
+                                ModificationUtil.AddChildAfter(insertedNewLine, binding)
+                            else
+                                ModificationUtil.AddChildAfter(anchor, binding)
+
+                        resultingNode
+
+                indexedBindings[fieldIndex] <- actualBinding
+                generatedBindings.AddLast(actualBinding) |> ignore
+            else
+                if generateSingleLine && isNull createdBinding.Semicolon then
+                    addSemicolon createdBinding
+
+        generatedBindings
+
+    member private x.SkipComments (node: ITreeNode): ITreeNode =
+        let mutable current = node
+        let mutable found = false
+
+        while not found do
+            let next = current.NextSibling
+            if next == null then found <- true
+
+            else if next.GetTokenType() == FSharpTokenType.WHITESPACE then
+                let nextNext = next.NextSibling
+                if nextNext == null then found <- true
+
+                // Whitespace + Comment: set comment as current
+                else if nextNext.IsCommentToken() then
+                    current <- nextNext
+
+                else found <- true
+
+            else found <- true
+
+        current
+
+    member private x.AreBindingsOrdered (bindings: TreeNodeCollection<IRecordFieldBinding>)
+        (declaredFields: string array) =
+        if declaredFields.Length <= 1 then true else
+
+        let mutable declaredFieldIndex = 0
+        let mutable bindingIndex = 0
+        let mutable ordered = true
+
+        while bindingIndex < bindings.Count && ordered do
+            while declaredFieldIndex < declaredFields.Length &&
+                  declaredFields[declaredFieldIndex] != bindings[bindingIndex].ReferenceName.ShortName do
+                declaredFieldIndex <- declaredFieldIndex + 1
+
+            if declaredFieldIndex >= declaredFields.Length then
+                ordered <- false
+
+            bindingIndex <- bindingIndex + 1
+            declaredFieldIndex <- declaredFieldIndex + 1
+
+        ordered
+
+    member private x.CreateOrderedIndexedBindings (bindings: IRecordFieldBinding[])
```

Comment:
Not critical: it'd be probably be better to define it as a function instead of a curried method? The function would automatically become private and it's much more common to have curried functions than methods.